storm-client/src/jvm/org/apache/storm/messaging/netty/Client.java

```diff
@@ -82,6 +82,10 @@
     private final ClientBootstrap bootstrap;
     private final InetSocketAddress dstAddress;
     protected final String dstAddressPrefixedName;
+    //The actual name of the host we are trying to connect to so that
+    // when we remove ourselves from the connection cache there is no concern that
+    // the resolved host name is different.
+    private final String host;
```

Comment:
Nit: Rename to dstHost or something like that to make it clear which host it is without having to read the comment.